In [7]:
import numpy as np
import pandas as pd

### Question 1: Give a count of all videos currently out.

```
Select count(rental_date)
from rental
where return_date is Null 
```

In [16]:
df1 =pd.read_csv('1.csv')
df1

number_of_videos_out
0                   183

***There are 183 videos that are currently out***

### Question 2: Make a list of all customer names who have videos out and how much they owe.
```
select al.customer_id, cus.first_name, cus.last_name, al.video_owe
from
	(select cus.customer_id,
			count(cus.customer_id) as video_owe
	from customer as cus
	inner join rental as r on r.customer_id = cus.customer_id
	where return_date is null
    group by 1) al
inner join customer as cus on cus.customer_id = al.customer_id
order by 4 desc;
```

In [19]:
df2 =pd.read_csv('2.csv')
df2.head()

customer_id first_name   last_name  video_owe
0           75      Tammy     Sanders          3
1          228    Allison     Stanley          2
2          361   Lawrence      Lawton          2
3          448     Miguel  Betancourt          2
4          163      Cathy     Spencer          2

***Above is the list of all customer names who have videos out and the number of video they owe, the person that owe the most is named Tammy Sanders***

### Question 5: What hour of the day to people rent the most ?
```
SELECT count(EXTRACT(hour from rental_date)) as Frequency, EXTRACT(hour from rental_date) as Rental_hour
FROM rental
group by rental_hour
order by 1 desc;
```


In [20]:
df3 =pd.read_csv('3.csv')
df3

frequency  rental_hour
0         887         15.0
1         696          8.0
2         694          0.0
3         688         18.0
4         684          3.0
5         681          4.0
6         676         19.0
7         673         10.0
8         671         21.0
9         667          7.0
10        664         16.0
11        663         11.0
12        658         20.0
13        653         14.0
14        652          9.0
15        649          1.0
16        648          5.0
17        647          6.0
18        645         13.0
19        642         23.0
20        634         17.0
21        632         12.0
22        630          2.0
23        610         22.0

***From the result, we know 3pm in the afternoon is the hour of day that people rent the most***

### Question 6: Which store is more profitable, assuming all movies cost 15 per inventory item to purchase.
```
select inv.store_id, count(inv)*15 as Cost, sum(lol.i_income) as Revenue, (sum(lol.i_income) - count(inv)*15) as Profit
from 
(SELECT rental.inventory_id, sum(payment.amount) as i_income
FROM payment
inner join rental on rental.rental_id = payment.rental_id
GROUP BY rental.inventory_id
order by 1) lol
inner join inventory as inv on inv.inventory_id = lol.inventory_id
group by inv.store_id
order by 1;
```

In [11]:
df4=pd.read_csv('4.csv')
df4

store_id   cost   revenue   profit
0         1  34050  30628.91 -3421.09
1         2  34650  30683.13 -3966.87

***As of right now, neither of the store are making profit becasue there is missing data (have rental_id but not payment_id) and the inventory that is purchased by the store but not been rented out at all.  Based on this result, store 1 is more profitable than store 2.***